### PR TITLE
Make validator app-operator version extraction on par with mutator

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: pkg/app/mutate_app.go [hunk 1]
```diff
@@ -96,36 +108,23 @@ func (m *Mutator) MutateApp(ctx context.Context, app v1alpha1.App) ([]mutator.Pa
...
+		m.logger.Debugf(ctx, "skipping mutation of app %#q in namespace %#q due to missing app-operator version label", app.Name, app.Namespace)
 		return nil, nil
+	} else if err != nil {
+		return nil, microerror.Mask(err)
 	}
 
 	// If the app CR does not have the unique version and is < 3.0.0 we skip
 	// the defaulting logic. This is so the admission controller is not enabled
 	// for existing platform releases.
-	if key.VersionLabel(app) != uniqueAppCRVersion && ver.Major() < 3 {
```
Comment (kopiczko) [inline pkg/app/mutate_app.go:123]: `0.0.0` major is < 3 so I removed that check.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fail when mutation review request fails. So far mutation review failures were
   ignored.
 
+### Fixed
+
+- Fall back to reconciling app-operator version from chart-operator App CR in
+  in validation.
```
Comment (JosephSalisbury) [inline CHANGELOG.md:18]:
```suggestion
  validation.
```

File: pkg/app/mutate_app.go (hunk 1 above)
Comment (kopiczko) [reply pkg/app/mutate_app.go:123]: I'm stupid that means that it will work for MC apps. I'm reverting this.

**Review by kubasobon — APPROVED** [review]

**Review by rossf7 — APPROVED** [review]
LGTM when tested

**Comment by kopiczko** [comment]
BTW @rossf7 are there cases when `app-operator.giantswarm.io/version` label is not set on an App CR?

**Comment by rossf7** [comment]
> Are there cases when app-operator.giantswarm.io/version label is not set on an App CR?

Yes when customers are creating app CRs directly via the MC API. There is an example in the docs PR. https://github.com/giantswarm/docs/pull/731

Currently in our docs the version label is always 1.0.0 but sometimes users forget and keeping the docs up to date with the version of app-operator is not practical. So we want to default it.

**Comment by kopiczko** [comment]
Thank you for your effort put into reviewing it. I decided to focus on kubebuilder (also for webhooks) instead.

**Pull request closed without merging**